models/classes/LtiProvider/ConfigurableLtiProviderRepository.php

```diff
@@ -93,4 +93,18 @@ private function getProviders()
 
         return $this->providers;
     }
+
+    /**
+     * @param string $id
+     * @return LtiProvider
+     */
+    public function searchById($id)
+    {
+        return current(array_filter(
```

Comment:
Opinion about using a `foreach` with a return as soon as the provider is found, instead of filtering?
```
foreach ($this->getProviders() as $provider) {
    if ($provider->getId() === $id) {
        return $provider;
    }
}
```

Reply:
Nice Idea, I'll make changes really soon;